Modelica/Clocked/UsersGuide/Clocks.mo

```diff
@@ -49,44 +74,61 @@ is generated:
 </div>
 
 <p>
-As usual in synchronous languages, a clock is represented by a <strong>true</strong> value
-when the clock is active. The relationship between such derived
+The relationship between such derived
 clocks is <strong>exact</strong>, so it is guaranteed that at every 3rd tick of clock
 \"periodicRealClock.y\", the clock \"subSample.y\" is active.
 </p>
 
+<h4>The two types of clocked partitions</h4>
+<p>
+A clocked partition is a set of equations that depend
+on each other and where the boundary variables are marked
+with sample and hold operators (boundary with other clocked partitions
+or with the global continuous-time partition).
+</p>
+<p>
+There are two types of clocked partitions:
+</p>
+<ul>
+    <li>A <strong>clocked discrete-time</strong> partition shall contain
+    <em>no operator relating to continuous time</em>:
+    <strong>der</strong>, <strong>delay</strong>, <strong>spatialDistribution</strong>,
+    no event-related operators (with exception of <strong>noEvent</strong>(&hellip;)),
+    and no <strong>when</strong>-clause with a Boolean condition.
+    A clocked discrete-time partition is thus a standard sampled data system described by difference equations
+    (it can typically contain the <strong>previous</strong> operator).</li>
+    <li>A <strong>clocked discretized continuous-time</strong> partition shall contain
+    neither operator <strong>previous</strong> nor operator <strong>interval</strong> (i.e. discrete-time operators).
+    It can contain continuous time operators (<strong>der</strong>&hellip;) and it has to be solved with a <strong>solver</strong> (see below).
+    </li>
+</ul>
+<p>
+Notice that it is an error if a partition doesn't fall into one of these two categories, e.g., if operators
+<strong>previous</strong> and <strong>der</strong> are both used in the same partition.
+This means it is <em>not allowed to mix continuous time and discrete-time operators</em> in a clocked partition
+(unlike in the global continuous time partition).
+</p>
+<p>
+Also notice that in a clocked discrete-time partition all event-generating mechanisms
+do no longer apply. Especially neither relations, nor one of the built-in event
+triggering operators will trigger an event.
+</p>
+
+<h4>Solver for clocked discretized continuous-time partitions</h4>
 <p>
 If a clock is associated to a clocked continuous-time partition, then an <strong>integrator</strong>
 has to be defined that is used to integrate the partition from the previous
-to the current clock tick. This is performed by setting parameter <strong>useSolver</strong>
-= <strong>true</strong> and defining the integration method as String with
-parameter <strong>solver</strong>. Both parameters are in tab <strong>Advanced</strong>
-of one of the clock signal generating blocks.
+to the current clock tick.
```

Comment:
Technically this is now only needed if the sub-clock partition has continuous-time equations; https://github.com/modelica/ModelicaSpecification/pull/3090 but likely to esoteric to mention.